dcos/package.py

```diff
@@ -24,28 +24,29 @@ def uninstall(pkg, package_name, remove_all, app_id, cli, app):
     :rtype: None
     """
 
-    if cli is False and app is False:
-        cli = app = True
-
     uninstalled = False
     installed = installed_packages(
-        pkg, app_id, package_name, cli_only=False)
-    installed_cli = next((True for installed_pkg in installed
-                          if installed_pkg.get("command")), False)
-    installed_app = next((True for installed_pkg in installed
-                          if installed_pkg.get("apps")), False)
+        pkg, None, package_name, cli_only=False)
+    installed_pkg = next(iter(installed), None)
+    installed_cli = installed_pkg.get("command") if installed_pkg else None
+    installed_app = installed_pkg.get("apps") if installed_pkg else []
 
-    if cli and installed_cli:
-        if subcommand.uninstall(package_name):
-            uninstalled = True
+    if cli is False and app is False:
+        app = True
+
+        # remove CLI only if there would be no remaining apps after uninstall
+        cli = (remove_all or not app_id
+                or installed_app.count(app_id) == len(installed_app))
 
     if app and installed_app:
         if pkg.uninstall_app(package_name, remove_all, app_id):
             uninstalled = True
```

Comment:
I think we should have separate variables for `app_ininstalled` and `cli_uninstalled` so we can print a more useful error message about what was / wasn't uninstalled if there was an error.

Also, I don't know if this is possible, but we should print a warning if the user requested to remove the CLI, but we didn't because there is an app still installed that relies on it.

Reply:
if the user requests to remove the CLI it gets removed, the "smart" logic only applies when there no `--cli` or `--app` flags.

Reply:
Also, to clarify, using `--cli` and `--app` at the same time is unsupported (it will trigger an invalid usage error, they are defined as mutually exclusive options).

Reply:
That in itself seems odd. So passing `--app` alone will only uninstall the app. Passing `--cli` alone will only uninstall the CLI. And passing neither will uninstall both.
...
But passing both installs neither?

I would think that passing both should also uninstall both.

Reply:
I've moved the double false check at the command level, and use the None value for the cli param, cf. last commit.